dig.go

```diff
@@ -479,6 +519,27 @@ func (c *Container) Invoke(function interface{}, opts ...InvokeOption) error {
 	return nil
 }
 
+// Child returns a named child of this container. The child container has
+// full access to the parent's types, and any types provided to the child
+// will be made available to the parent.
+//
+// The name of the child is for observability purposes only. As such, it
+// does not have to be unique across different children of the container.
+func (c *Container) Child(name string) *Container {
+	child := &Container{
+		providers: make(map[key][]*node),
+		values:    make(map[key]reflect.Value),
+		groups:    make(map[key][]reflect.Value),
+		rand:      rand.New(rand.NewSource(time.Now().UnixNano())),
```

Comment:
We shouldn't instantiate a new source of randomness. We should use the
parent's `rand` so that the behavior of the child is deterministic in tests.

```suggestion
		rand:      c.rand,
```